replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/QueryServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package org.codice.ditto.replication.api.impl;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.codice.ditto.replication.api.QueryService;
+import org.codice.ditto.replication.api.ReplicationException;
+import org.codice.ditto.replication.api.data.Query;
+import org.opengis.filter.Filter;
+
+public class QueryServiceImpl implements QueryService {
+
+  private final CatalogFramework framework;
+
+  private final FilterBuilder builder;
+
+  private final InputTransformer xmlTransformer;
+
+  public QueryServiceImpl(
+      CatalogFramework framework, FilterBuilder builder, InputTransformer xmlTransformer) {
+    this.framework = framework;
+    this.builder = builder;
+    this.xmlTransformer = xmlTransformer;
+  }
+
+  @Override
+  public List<Query> getQueries() {
+    Filter filter = builder.attribute(Core.METACARD_TAGS).is().equalTo().text("workspace");
+    QueryResponse queryResponse;
+    try {
+      queryResponse = framework.query(new QueryRequestImpl(new QueryImpl(filter)));
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      throw new ReplicationException("Failed to retrieve Query metacards {}", e);
+    }
+
+    List<Query> queries = new ArrayList<>();
+    for (Result result : queryResponse.getResults()) {
+      Metacard mcard = result.getMetacard();
+      queries.addAll(getQueries(mcard));
+    }
+    return queries;
+  }
+
+  public List<Query> getQueries(Metacard metacard) {
+    List<Query> queries = new ArrayList<>();
+    List<Metacard> metacards =
+        getQueryStrings(metacard).stream().map(this::xmlToMetacard).collect(Collectors.toList());
```

Comment:
❓ Why collect just to iterate that temporary collection? Just continue streaming to perform what you have in the loop below. Suggestion would be to create a private method that accepts a `Metacard` and returns a `org.codice.ditto.replication.api.impl.data.QueryImpl` which would make the stream much more readable and this function much simpler.